internal/resources/fabric/precision_time/resource.go

```diff
@@ -0,0 +1,463 @@
+package precision_time
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"reflect"
+	"slices"
+	"time"
+
+	"github.com/equinix/equinix-sdk-go/services/fabricv4"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+
+	equinix_errors "github.com/equinix/terraform-provider-equinix/internal/errors"
+	"github.com/equinix/terraform-provider-equinix/internal/framework"
+)
+
+func NewResource() resource.Resource {
+	return &Resource{
+		BaseResource: framework.NewBaseResource(
+			framework.BaseResourceConfig{
+				Name: "equinix_fabric_precision_time",
+			},
+		),
+	}
+}
+
+type Resource struct {
+	framework.BaseResource
+}
+
+func (r *Resource) Schema(
+	ctx context.Context,
+	req resource.SchemaRequest,
+	resp *resource.SchemaResponse,
+) {
+	resp.Schema = resourceSchema(ctx)
+}
+
+func (r *Resource) Create(
+	ctx context.Context,
+	req resource.CreateRequest,
+	resp *resource.CreateResponse,
+) {
+
+	var plan PrecisionTimeModel
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Retrieve the API client from the provider metadata
+	client := r.Meta.NewFabricClientForFramework(ctx, req.ProviderMeta)
+
+	createRequest, diags := buildCreateRequest(ctx, plan)
+
+	resp.Diagnostics.Append(diags...)
+	if diags.HasError() {
+		return
+	}
+
+	ept, _, err := client.PrecisionTimeApi.CreateTimeServices(ctx).PrecisionTimeServiceRequest(createRequest).Execute()
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error creating Precision Time Service",
+			equinix_errors.FormatFabricError(err).Error(),
+		)
+		return
+	}
+
+	createTimeout, diags := plan.Timeouts.Create(ctx, 10*time.Minute)
+	if diags.HasError() {
+		resp.Diagnostics.Append(diags...)
+		return
+	}
+	createWaiter := getCreateUpdateWaiter(ctx, client, ept.GetUuid(), createTimeout)
+	eptChecked, err := createWaiter.WaitForStateContext(ctx)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			fmt.Sprintf("Failed to create Precision Time Service %s", ept.GetUuid()), err.Error())
+		return
+	}
+
+	// Parse API response into the Terraform state
+	resp.Diagnostics.Append(plan.parse(ctx, eptChecked.(*fabricv4.PrecisionTimeServiceCreateResponse))...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Set state to fully populated data
+	resp.Diagnostics.Append(resp.State.Set(ctx, &plan)...)
```

Comment:
Is there a risk of overwriting API response data with corresponding plan data here?  I'm assuming that any intersections between request data fields and response data fields are expected to have the same value, but is that guaranteed to be the case?